Bump home page to 2.3 blog

seemethere opened the pull request:
Bumps the home page to reference the 2.3 blog

gchanan commented:
looks pretty good, a couple of nits:

- default to phrasing things positively instead of negatively ("native integration" or "improved performance") or something.
- better to include 2 things, you could say something like "user-defined triton kernel performance improvements, TensorParallelism API improvements"

seemethere commented:
> looks pretty good, a couple of nits:
> 
> * default to phrasing things positively instead of negatively ("native integration" or "improved performance") or something.
> * better to include 2 things, you could say something like "user-defined triton kernel performance improvements, TensorParallelism API improvements"

Updated to include TensorParallelism and removed negative language!
